### PR TITLE
4 get api reviews

### PR DESCRIPTION
**Pull request by LKFULLER1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -41,17 +39,20 @@ describe('4 - GET /api/reviews', () => {
...
-            expect.objectContaining({
-              slug: expect.any(String),
-              description: expect.any(String)
-
-
-              
-            })
-          );
+          expect(review).toMatchObject({
+            title: expect.any(String),
```
Comment (mezzDavies) [inline __tests__/app.test.js:61]: Please double check what the ticket asks for to be returned and check for it here - there are properties that need adding and some that need removing from the assertion.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -41,17 +39,20 @@ describe('4 - GET /api/reviews', () => {
...
+            category: expect.any(String),
+            category: expect.any(String),
+            designer: expect.any(String),
+            owner: expect.any(String),
+            review_body: expect.any(String),
+            review_id: expect.any(Number),
+            review_img_url: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+          });
```
Comment (mezzDavies) [inline __tests__/app.test.js:71]: We also need check that the objects in the array are sorted by date in descending order.

File: models/games.js [hunk 3]
```diff
@@ -3,7 +3,13 @@ const db = require('../db/connection.js');
 exports.selectCategories = () => {
     return db.query('SELECT * FROM categories;')
         .then(data => {
-            console.log(data.rows)
+            return data.rows;
+        })
+};
+
+exports.selectReviews = () => {
+    return db.query('SELECT * FROM reviews;')
```
Comment (mezzDavies) [inline models/games.js]: This returns all columns from the reviews table but the ticket asks for so this query needs work to get all the required columns.

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Hi, Please have a look at the the properties that are required on each object that this endpoint sends back - it's not as simple as selecting everything from the reviews table unfortunately. I have left some comments that should point you in the right direction 👍

File: models/games.js (hunk 3 above)
Comment (LKFULLER1) [reply models/games.js]: I realised I hasn't read the ticket and have now implemented much more of the task, just stuck on a toBeSortedBy test - saying it is not a function

File: __tests__/app.test.js (hunk 1 above)
Comment (mezzDavies) [reply __tests__/app.test.js:61]: Well done - comment_count added. Also the category property is repeated in this assertion - one of them can be removed.

File: package.json [hunk 4]
```diff
@@ -30,7 +30,8 @@
   "devDependencies": {
     "husky": "^7.0.0",
     "jest": "^27.5.1",
-    "jest-extended": "^2.0.0"
+    "jest-extended": "^2.0.0",
+    "jest-sorted": "^1.0.14"
   },
   "jest": {
     "setupFilesAfterEnv": [
```
Comment (mezzDavies) [inline package.json:37]: Add jest-sorted to this array and then that should fix your "not a function" problem.

**Review by mezzDavies — APPROVED** [review]
Good stuff. make sure you see the note to hopefully help you get jest-sorted working.

**Pull request merged**